Added useSettingsGradle property in kotlin-client.

dylan-kwon opened the pull request:
#14699

Since the Arctic Fox version of Android Studio, projects using `settings.gradle` will throw an error.

`Build was configured to prefer settings repositories over project repositories but repository 'maven' was added by build file '{module_name}/build.gradle'`

Therefore, we added a property that can selectively add or remove problematic code lines like `omitGradleWrapper`.

The code is part of `build.gradle.mustache`.
```gradle
{{^useSettingsGradle}}
repositories {
     maven { url "https://repo1.maven.org/maven2" }
}
{{/useSettingsGradle}}
```

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@jimschubert, @dr4ke616 @karismann @Zomzog @andrewemery @4brunu @yutaka0m

4brunu commented:
Hi, thanks for creating this PR.
Could you please run the following commands and commit the changes related to kotlin?
```
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
```
Thanks

dylan-kwon commented:
Hi @4brunu, I updated the sample and docs.

While updating the docs, I noticed that `kotlinx_serialization` was missing from the description of `serializationLibrary` and added it.

Since `kotlinx_serialization` is already a library used by many kotlin-based projects, it will be helpful for those who gave up thinking that this library is not supported.

4brunu commented on modules/openapi-generator/src/main/resources/kotlin-client/build.gradle.mustache:
```diff
@@ -54,10 +54,12 @@ apply plugin: 'kotlinx-serialization'
 apply plugin: 'idea'
 {{/idea}}
 apply plugin: 'maven-publish'
+{{^useSettingsGradle}}
```
@dylan-kwon shouldn't we also remove the settings.gradle file?

dylan-kwon replied:
The `settings.gradle` I am talking about in this PR is `settings.gradle` in the root of the Gradle project that uses `open-apiGenerator`, not `settings.gradle` in a directory automatically generated by `open-apiGenerator`.

4brunu replied:
Thanks 👍
